### PR TITLE
Fix batch geocoder by polling state till completion

### PR DESCRIPTION
**Pull request by rafatower**
@Kartones @juanignaciosl please review

cc/ @javisantana @saleiva , FYI

Tested locally.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4306/))

**Comment by juanignaciosl** [comment]
:+1: as a quick fix, loop is quite confusing.

**Comment by javisantana** [comment]
could that loop be looping forever? It's ok as hotfix but i'd review that just in case

**Comment by rafatower** [comment]
sorry I should have some comments there:

- there's a timeout to prevent it for looping forever. I set it to 5h because don't really know how long it can take
- I recovererd old code that we agreed to remove because it's ugly and thought it wasn't needed
- it's not the best place for the polling, but it is needed.

File: app/models/geocoding.rb [hunk 1]
```diff
@@ -124,7 +132,15 @@ def run_geocoding!(processable_rows, rows_geocoded_before = 0)
     table_geocoder.run
 
     self.update remote_id: table_geocoder.remote_id
-    self.update(table_geocoder.update_geocoding_status)
+
+    # INFO: this loop polls for the state of the table_geocoder batch process and cannot be simply removed
+    begin
+      self.update(table_geocoder.update_geocoding_status)
+      raise 'Geocoding timeout' if Time.now - @started_at > run_timeout and ['started', 'submitted', 'accepted'].include? state
```
Comment (Kartones) [inline app/models/geocoding.rb]: if you use here `@run_timeout` you can avoid the `attr_accesor` above, but is ok this way also

**Comment by javisantana** [comment]
oh your right, I was confused by the [wrong] ruby way to place the ifs :D #blamingothers

**Comment by Kartones** [comment]
:+1:

**Pull request merged**